### ISSUE TITLE
Invalid Redirect Error When logging in, MPOS runnin on VM behind port forwarding

### ISSUE THREAD

**Issue by theneonirvana** [MPOS/php-mpos#2660]
Hi there...

Im running a couple sites that are open source off git hub... these sites reside on a Virtual Machine on a server

I have subdomains setup to forward to the website IP address and a diffrent port: for instance... subdomain1.website.com ---->> <websiteip>:82 

I then have portforwarding setup so that port 82 routes to the virtual machines ip address on port 80 where the website is hosted for example <wanip>:82 -->> <vmIP>:80

My problem is that the website has a login feature, when users click login it loads and then presents the following error:

Google Chrome users get "website has sent an Invalid Response"   INVALID_REDIRECT_ERR
Firefox Users get : " Corrupted Content Error" The site at http://WEBSITEIP:82/index.php?page=login has experienced a network protocol violation that cannot be repaired.
IE Users get : "Cannot Reach this page"

If you press F5 or hit refresh the page comes right up and your logged in correctly, but normal users dont know to hit F5, and this is a problem

i have tested this with the sites internally on the network, and everything works fine on all 3 browsers, the moment that you make requests from wan addresses it messes up. 

I have even tested this on a direct connection to a VM machine and it is working correctly, so i believe the problem is with the redirect entirely. 

The Virtual Machines are running Ubuntu 16 and apache2 and there sites are setup properly for virtual hosts on the correct ports.

I think it has somthing to do with HTTPS, but im not running ssl and dont know if i can turn it off in mpos

will setting up https? with a cert and everything fix this?

TL:DR : PHP pages gives error when logging in, is fine if you press f5, works fine behind lan, dosnt work behind wan with routing to VM's

**Comment by Kopernik999** [comment]
I have exactly the same problem, if you have a solution please let me know

**Comment by theneonirvana** [comment]
Nope, i tried everything aside from buying a ssl ticket, and dont even know if that would resolve it. 

So i had to make a work around,  basically putting MPOS on a remote exposed box with a static ip, and doing subdomains to that box. 

Sorry

**Comment by Kopernik999** [comment]
Okay - I'll try to find a solution - I think it's not SSL Problem is it a CSRF and the Memcache administration Problem. I keep you updated

**Comment by TheSerapher** [comment]
Just a heads up, development on MPOS was done in a virtual box without SSL which requires Port Forwarding. I am guessing since Source and destination port aren’t the same you run into these issues.

Maybe the redirect after login tries to reach a different port than expected ...

You’ll have to dig into this but I’d think it’s an easy solution somewhere. If you find a fix feel free to post it.

**Comment by Kopernik999** [comment]
Hello,

Here is my solution that works wonderfully.

Just change the port in user.class.php line 650 and in login.inc.php in line 30.
(In your case from 80 -> 82). I've still removed the http (s) for security in the lines.

Kind regards

**Comment by theneonirvana** [comment]
Wow! Thats a great fix,    22 days later :(   

Least i know for next time....   and if i have a problem with a coin now with remote connections...

Jesus,  as much as that makes me happy it makes me frustrated, no one could point this out to me a month ago....

:(

**Comment by TheSerapher** [comment]
That’s how it is with unsupported software ;-)

**Comment by KevinBiomech** [comment]
The greatest strength of open source software is that anyone can contribute.

The greatest weakness of open source software is that anyone can contribute.
